sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/InvocationVisitor.cs

```diff
@@ -107,14 +107,21 @@ private ProgramState HandleStringNullOrWhiteSpaceCheckMethod()
         {
             var newProgramState = this.programState
                 .PopValue(out var arg1)
-                .PopValue();
+                .PopValue(out var arg2);
 
-            // Handle string.IsNullOrWhiteSpace(arg1) as if it was arg1 == null
-            return new ReferenceEqualsConstraintHandler(arg1, SymbolicValue.Null,
-                    this.invocation.ArgumentList.Arguments[0].Expression,
-                    null,
-                    newProgramState, this.semanticModel)
-                .PushWithConstraint();
+            var refEquals = new ReferenceEqualsSymbolicValue(arg1, arg2);
+            newProgramState = newProgramState.PushValue(refEquals);
+            if (newProgramState.HasConstraint(arg1, ObjectConstraint.Null) ||
+                    newProgramState.HasConstraint(arg1, StringConstraint.EmptyString) ||
+                    newProgramState.HasConstraint(arg1, StringConstraint.WhiteSpaceString))
+            {
+                newProgramState = newProgramState.SetConstraint(refEquals, BoolConstraint.True);
+            }
+            else if (newProgramState.HasConstraint(arg1, StringConstraint.FullNotWhiteSpaceString))

```

Comment:
add a comment about `FullOrNullString`, that we don't know the state so we don't treat it here